File: SiliconLabsApp/ViewControllers/AppSelection/SILDemoNavigationViewController.m
//
//  SILDemoNavigationViewController.m
//  BlueGecko
//
//  Created by Kamil Czajka on 24/01/2020.
//  Copyright © 2020 SiliconLabs. All rights reserved.
//

#import "SILDemoNavigationViewController.h"
#import "SILAppSelectionViewController.h"

@interface SILDemoNavigationViewController () <UINavigationControllerDelegate>

@end

@implementation SILDemoNavigationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.delegate = self;
}

- (void)navigationController:(UINavigationController *)navigationController willShowViewController:(UIViewController *)viewController animated:(BOOL)animated {
    if ([viewController isKindOfClass:[SILAppSelectionViewController class]]) {
        SILAppSelectionViewController* demoController = viewController;
        demoController.appsArray = [SILApp demoApps];
    }
}

@end

File: SiliconLabsApp/ViewControllers/AppSelection/SILDemoNavigationViewController.swift
//
//  SILDemoNavigationViewController.swift
//  BlueGecko
//
//  Created by Grzegorz Janosz on 13/05/2021.
//  Copyright © 2021 SiliconLabs. All rights reserved.
//

import Foundation

class SILDemoNavigationViewController: UINavigationController, UINavigationControllerDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        delegate = self
    }

    func navigationController(_ navigationController: UINavigationController, willShow viewController: UIViewController, animated: Bool) {
        if let demoController = viewController as? SILAppSelectionViewController, let demoApps = SILApp.demoApps() as? [SILApp] {
            demoController.appsArray = demoApps
        }
    }
}
